lib/cartopy/io/ogc_clients.py

```diff
@@ -398,6 +401,18 @@ def __init__(self, wmts, layer_name, gettile_extra_kwargs=None):
 
         self._matrix_set_name_map = {}
 
+        # Enable a cache mechanism when cache is equal to True or to a path.
+        self._default_cache = False
+        if cache is True:
+            self._default_cache = True
+            self.cache_path = cartopy.config["cache_dir"]
+        elif cache is False:
+            self.cache_path = None
+        else:
+            self.cache_path = cache
```

Comment:
We should stick with Pathlib objects here if we can which will make things easier later on.
```suggestion
            self.cache_path = Path(cache)
```